fix(release): set s-b version info for CI built release

dimakr opened the pull request:
The change updates the release CI workflow to ensure that s-b and scylla-gocql version info includes commit date and SHA attributes.

Fixes: https://github.com/scylladb/scylla-bench/issues/178

dimakr commented:
Some testing of the change:
- to simulate executing s-b CI release workflow locally, the `act` util was used.
A local test tag was created, for GoReleaser to use this tag.
Additionally the `Run GoReleaser` step of release workflow was executed with skipping the publish
```
args: release --clean --skip=publish 
```
to avoid publishing test artifacts. This goreleaser option was removed before pushing the change to review.

The result of executing CI release workflow locally:
```
❯ act --secret-file .secrets -P ubuntu-24.04=ghcr.io/catthehacker/ubuntu:act-latest -W .github/workflows/release.yml -j release
INFO[0000] Using docker host 'unix:///var/run/docker.sock', and daemon socket 'unix:///var/run/docker.sock' 
...
[Release scylla-bench/release] ⭐ Run Main Extract scylladb-gocql driver info
[Release scylla-bench/release]   🐳  docker exec cmd=[bash -e /var/run/act/workflow/5] user= workdir=
| Looking for scylla-gocql driver version...
| Driver version: v1.14.5
| Driver repository: github.com/scylladb/gocql (owner: scylladb)
| Trying GitHub API for version info...
| Using GitHub token for API requests
| Driver version info:
|   Version: v1.14.5
|   Commit:  597d7e3f74ebb11c401a18c26ff372b6694e7037
|   Date:    2025-01-26T00:48:29Z
[Release scylla-bench/release]   ✅  Success - Main Extract scylladb-gocql driver info [948.653281ms]
[Release scylla-bench/release]   ⚙  ::set-env:: DRIVER_VERSION=v1.14.5
[Release scylla-bench/release]   ⚙  ::set-env:: DRIVER_COMMIT=597d7e3f74ebb11c401a18c26ff372b6694e7037
[Release scylla-bench/release]   ⚙  ::set-env:: DRIVER_DATE=2025-01-26T00:48:29Z
[Release scylla-bench/release] ⭐ Run Main Run GoReleaser
[Release scylla-bench/release]   🐳  docker cp src=/home/dmitriy/.cache/act/goreleaser-goreleaser-action@v6/ dst=/var/run/act/actions/goreleaser-goreleaser-action@v6/
[Release scylla-bench/release]   🐳  docker exec cmd=[/opt/acttoolcache/node/18.20.8/x64/bin/node /var/run/act/actions/goreleaser-goreleaser-action@v6/dist/index.js] user= workdir=
| Downloading https://github.com/goreleaser/goreleaser/releases/download/v2.4.5/goreleaser_Linux_x86_64.tar.gz
| Extracting GoReleaser
| [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /tmp/a27f3d81-a50d-400e-9a53-15e5e04ae96a -f /tmp/29a33cec-8ca7-42ad-a686-c26727bed5f9
| GoReleaser v2.4.5 installed successfully
| [command]/opt/hostedtoolcache/goreleaser-action/2.4.5/x64/goreleaser release --clean --skip=publish --skip=validate
|   • skipping announce, publish and validate...
|   • cleaning distribution directory
|   • loading environment variables
|     • using token from  $GITHUB_TOKEN 
|   • getting and validating git state
|     • git state                                      commit=5127a654905d7cd9b155c9256f1c9c56e04a2281 branch=HEAD current_tag=v0.0.1-test previous_tag=v0.2.0 dirty=true
|     • pipe skipped                                   reason=validation is disabled
|   • parsing tag
|   • setting defaults
|   • ensuring distribution directory
|   • setting up metadata
|   • writing release metadata
|   • loading go mod information
|   • build prerequisites
|   • building binaries
|     • building                                       binary=dist/scylla-bench_linux_arm64_v8.3/scylla-bench
|     • building                                       binary=dist/scylla-bench_linux_amd64_v3/scylla-bench
|   • generating changelog
|   • archives
|     • creating                                       archive=dist/scylla-bench_0.0.1-test_linux_arm64.tar.gz
|     • creating                                       archive=dist/scylla-bench_0.0.1-test_linux_amd64.tar.gz
|   • creating source archive
|     • creating source archive                        file=dist/scylla-bench-0.0.1-test.tar.gz
|   • calculating checksums
|   • size reports
|     • 8.5MiB                                         path=dist/scylla-bench_linux_amd64_v3/scylla-bench
|     • 8MiB                                           path=dist/scylla-bench_linux_arm64_v8.3/scylla-bench
|     • 3.172MiB                                       path=dist/scylla-bench_0.0.1-test_linux_arm64.tar.gz
|     • 3.481MiB                                       path=dist/scylla-bench_0.0.1-test_linux_amd64.tar.gz
|   • docker images
|     • building docker image                          image=scylladb/scylla-bench:0.0.1-test-arm64v8
|     • building docker image                          image=scylladb/scylla-bench:0.0.1-test-amd64
|   • writing artifacts metadata
|   • release succeeded after 2s
|   • thanks for using goreleaser!
...
[Release scylla-bench/release] Cleaning up container for job release
[Release scylla-bench/release]   ✅  Success - Complete job
[Release scylla-bench/release] 🏁  Job succeeded
```
Using the built image - now it contains the commit date and SHA portions in version data:
```
❯ docker run --rm scylladb/scylla-bench:0.0.1-test-amd64 -version-json
{
  "scylla-bench": {
    "version": "0.0.1-test",
    "commit_date": "2025-04-21T14:06:06Z",
    "commit_sha": "5127a654905d7cd9b155c9256f1c9c56e04a2281"
  },
  "scylla-driver": {
    "version": "v1.14.5",
    "commit_date": "2025-01-26T00:48:29Z",
    "commit_sha": "597d7e3f74ebb11c401a18c26ff372b6694e7037"
  }
}
```
- regression check to ensure that for the case of building s-b locally the version info is extracted correctly
```
❯ go build -o scylla-bench
❯ ./scylla-bench -version-json
{
  "scylla-bench": {
    "version": "v0.2.0-dev-45ccc40b",
    "commit_date": "2025-04-21T16:33:11Z",
    "commit_sha": "45ccc40b28b91274bedf3449954b11cd29a6c812"
  },
  "scylla-driver": {
    "version": "v1.14.5",
    "commit_date": "2025-01-26T00:48:29Z",
    "commit_sha": "597d7e3f74ebb11c401a18c26ff372b6694e7037"
  }
}
```

vponomaryov commented on internal/version/version.go:
```diff
@@ -212,6 +216,15 @@ func extractRepoOwner(repoPath, defaultOwner string) string {
 // Extracts driver version info
 func getDriverVersionInfo() ComponentInfo {
+	// check if driver version info was set via ldflags
+	if driverVersion != "unknown" {
```
What about checking `driverCommit` and `driverDate`?

dimakr replied:
This is basically a gating kind of check if a version info was set via ldflags (i.e. on build stage). If version is set, we use other components as they were set.
Otherwise we would need to get all the info from scratch in runtime.

vponomaryov replied:
Actually I was referring to the fact that we check just 1 var but use 3.
But, probably, not a big deal.

vponomaryov commented on scripts/extract-driver-version.sh:
```diff
@@ -0,0 +1,39 @@
+set -e
+
+echo "Looking for scylla-gocql driver version..." >&2
+DRIVER_VERSION=$(go list -m -json all | grep -A 10 "github.com/scylladb/gocql" | grep '"Version"' | head -1 | cut -d'"' -f4)
```
Ok

vponomaryov reviewed (approved):
LGTM

vponomaryov reviewed (approved):
LGTM

fruch commented:
@dimakr 

See if you can push a tag here and release a new version

dimakr commented:
> > See if you can push a tag here and release a new version
> 
> @fruch tried triggering new release and failed on release workflow, added a small fix #180 for that (didn't run into that issue initially as during testing I disabled pushing phase in my local CI release workflow runs, and that probably made GoReleaser ignore git dirty state checks)

@fruch now after that last minor fix I was able to trigger the release, added a new v0.2.2 one
